components/sidebar/sidebar_category_header.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {DraggableProvidedDragHandleProps} from 'react-beautiful-dnd';
+
+import {wrapEmojis} from 'utils/emoji_utils';
+
+type Props = {
+    children?: React.ReactNode;
+    displayName: string;
+    dragHandleProps?: DraggableProvidedDragHandleProps;
+    isCollapsed?: boolean;
+    isCollapsible?: boolean;
+    isDragging?: boolean;
+    isDraggingOver?: boolean;
+    onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+}
+
+const SidebarCategoryHeader = React.forwardRef((props: Props, ref?: React.Ref<HTMLDivElement>) => {
+    return (
+        <div className='SidebarChannelGroupHeader'>
+            <div
```

Comment:
This has changed from a button to a clickable div. I hoped to switch between the two components based on whether or not the header should be clickable, but I ran into typing issues because of the ref. This seems to work well enough though since the header can still be tabbed through as before.